src/chocolatey.benchmark/helpers/PinvokeProcessHelper.cs

```diff
@@ -0,0 +1,472 @@
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.ConstrainedExecution;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+using System.Security.Permissions;
+using System.Security;
+using chocolatey.infrastructure.information;
+using chocolatey.infrastructure.platforms;
+
+namespace chocolatey.benchmark.helpers
+{
+    internal class PinvokeProcessHelper
+    {
+        private static readonly string[] _filteredParents = new[]
+        {
+            "explorer",
+            "powershell",
+            "pwsh",
+            "cmd",
+            "bash",
+            // The name used to launch windows services
+            // in the operating system.
+            "services",
+            // Known Terminal Emulators
+            "Tabby",
+            "WindowsTerminal",
+            "FireCMD",
+            "ConEmu64",
+            "ConEmuC64"
+        };
+
+        public static ProcessTree GetDocumentedProcessTree(Process process = null)
+        {
+            if (process == null)
+            {
+                process = Process.GetCurrentProcess();
+            }
+
+            var tree = new ProcessTree(process.ProcessName);
+
+            if (Platform.GetPlatform() != PlatformType.Windows)
+            {
+                return tree;
+            }
+
+            Process nextProcess = null;
+
+            while (true)
+            {
+                var foundProcess = ParentDocumentedHelper.ParentProcess(nextProcess ?? process);
+
+                if (foundProcess == null)
+                {
+                    break;
+                }
+
+                nextProcess = foundProcess;
+                tree.Processes.AddLast(nextProcess.ProcessName);
+            }
+
+            return tree;
+        }
+
+        public static string GetDocumentedParent(Process process = null)
+        {
+            if (Platform.GetPlatform() != PlatformType.Windows)
+            {
+                return null;
+            }
+
+            if (process == null)
+            {
+                process = Process.GetCurrentProcess();
+            }
+
+            Process nextProcess = null;
+
+            while (true)
+            {
+                var foundProcess = ParentDocumentedHelper.ParentProcess(nextProcess ?? process);
+
+                if (foundProcess == null)
+                {
+                    break;
+                }
+
+                nextProcess = foundProcess;
+            }
+
+            return nextProcess?.ProcessName;
+
+            //var parentProcess = ParentDocumentedHelper.ParentProcess(process);
+
+            //if (parentProcess == null)
+            //{
+            //    return null;
+            //}
+
+            //var topLevelProcess = GetDocumentedParent(parentProcess);
+
+            //if (topLevelProcess == null)
+            //{
+            //    return parentProcess.ProcessName;
+            //}
+
+            //return topLevelProcess;

```

Comment:
Are these comments needed?

Reply:
These comments are not needed anymore, and can be safely removed.